src/test/java/com/thealgorithms/bitmanipulation/BitSwapTest.java

```diff
@@ -0,0 +1,12 @@
+package com.thealgorithms.bitmanipulation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.Test;
+public class BitSwapTest {
+    @Test
+    void testHighestSetBit() {
+        assertEquals(3, BitSwap.bitSwap(3, 0, 1));
+        assertEquals(5, BitSwap.bitSwap(6, 0, 1));
+    }
```

Comment:
I would suggest to add a test case with equal positions.

```suggestion
        assertEquals(5, BitSwap.bitSwap(6, 0, 1));
        assertEquals(7, BitSwap.bitSwap(7, 1, 1));
    }
```